uwblueprint/planet-read
# Create development environment

## Notion ticket link

[Create development environment](https://www.notion.so/uwblueprintexecs/a658c933d18c48d4b43fd99bda2bd3b4?v=4a47edc475b34bb99b25ff1fae2a5507&p=2d4b16cf1a834269ba86ec183d058575)



## Implementation description
* Set up firebase frontend PR preview deployments
* Set up firebase frontend deployment on merges to main
* Set up heroku backend deployment on merges to main 



## Steps to test
1. Log in as Carl Sagan and poke around using the preview URL commented below to verify that requests to backend are going through properly (backend URL is https://uwbp-planet-read-preview.herokuapp.com/)
> NOTE: database was only partially seeded so issues with translations/contents not loading properly may be caused by that
2. Check that the staging URL also works as expected: https://planet-read-uwbp.web.app (backend URL is https://uwbp-planet-read.herokuapp.com/)
3. Automatic deployment to main can only be tested after this PR is merged



## What should reviewers focus on?
* Currently the Heroku backend deploys only from main
     * Heroku allows for apps to be deployed on PR; however, since the backend URL is currently set using GitHub environment secrets, the value for this secret would have to change with each PR if we wanted to have PR previews for both frontend/backend changes
* The database was seeded by running queries in the MySQL shell--would making a script to seed the JawsDB be useful? Or would seeding it properly once be fine?

## Checklist
